numpy/_core/src/umath/umathmodule.c

```diff
@@ -303,29 +304,32 @@ int initumath(PyObject *m)
      * TODO: This should probably be done at a better place, or even in the
      *       code generator directly.
      */
-    s = _PyDict_GetItemStringWithError(d, "logical_and");
-    if (s == NULL) {
+    int res = PyDict_GetItemStringRef(d, "logical_and", &s);
+    if (res <= 0) {
         return -1;
     }
     if (install_logical_ufunc_promoter(s) < 0) {
```

Comment:
You need Py_DECREF(s) in the error case, same below (in this file).

Reply:
thank you!